evalml/data_checks/invalid_target_data_check.py

```diff
@@ -243,6 +248,37 @@ def _check_target_has_nan(self, y, messages):
         elif null_rows.any():
             num_null_rows = null_rows.sum()
             pct_null_rows = null_rows.mean() * 100
+            rows_to_drop = null_rows.loc[null_rows].index.tolist()
+
+            action_options = []
+            impute_action_option = DataCheckActionOption(
```

Comment:
could we add these into the conditionals so we only run them each if the specified strategy is input?

Reply:
Sure, initially I had planned on having a `both` option but I think we can push that out.

Reply:
Neither of these are run during initialization of the `DataCheckActionOption`, it's only when they're specified in `action_options` for a `DataCheckError` that they get converted to a dictionary